src/hera/workflows/script.py

```diff
@@ -760,6 +755,27 @@ def script_wrapper(func: Callable[FuncIns, FuncR]) -> Callable:
         def task_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
             """Invokes a `Script` object's `__call__` method using the given SubNode (Step or Task) args/kwargs."""
             if _context.active:
+                if len(args) == 1 and isinstance(args[0], (InputV1, InputV2)):
```

Comment:
Note this new behaviour only affects calls made to a `@script`-decorated function when passed a Pydantic I/O object by argument, which previously would not have functioned meaningfully (the object was ignored).